src/dotty/tools/dotc/transform/PostTyper.scala

```diff
@@ -231,6 +231,15 @@ class PostTyper extends MacroTransform with IdentityDenotTransformer  { thisTran
                   ctx.compilationUnit.source.exists &&
                   sym != defn.SourceFileAnnot)
                 sym.addAnnotation(Annotation.makeSourceFile(ctx.compilationUnit.source.file.path))
+
+              for (parent <- sym.asClass.classInfo.classParents) {
+                val pclazz = parent.classSymbol
+                if (pclazz.is(Sealed)) {
+                  println(s"add $sym to $pclazz")
+                  pclazz.addAnnotation(Annotation.makeChild(sym))
+                }
+              }
+
```

Comment:
If a sealed trait has anonymous subclasses then it's impossible to pattern match on all subclasses. I think the simplest solution is to disallow anonymous subclasses of sealed traits.